core/src/main/java/io/kestra/core/models/executions/statistics/ExecutionCountStatistics.java

```diff
@@ -0,0 +1,49 @@
+package io.kestra.core.models.executions.statistics;
+
+import io.kestra.core.models.flows.State;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * ExecutionCounts.
+ *
+ * @param counts    The execution counts by state.
+ * @param total     The total execution count.
+ */
+public record ExecutionCountStatistics(
+    @NotNull Map<State.Type, Long> counts,
+    @NotNull Long total
+) implements Comparable<ExecutionCountStatistics> {
+
+    private static final Map<State.Type, Long> DEFAULT_COUNTS = Map.ofEntries(
```

Comment:
Maybe use the enum `values` list to automatically update the map initialization when a new status is added

Reply:
It was my idea at the start. But, it seems that all `State.Type` are not associated to an execution. For example, in the existing [`DailyExecutionStatistics`](https://github.com/kestra-io/kestra/blob/develop/core/src/main/java/io/kestra/core/models/executions/statistics/DailyExecutionStatistics.java#L26-L38) we don't use the type `RETRIED`. Is this an error or not ? Otherwise, I can just exclude that state.